src/network.rs

```diff
@@ -446,7 +565,7 @@ pub fn init(
 		.map(|text| PreSharedKey::from_str(&text))
 		.transpose()?;
 	// create transport
-	let transport = setup_transport(id_keys, psk, port_reuse);
+	let transport = setup_transport(id_keys.clone(), psk, port_reuse);
```

Comment:
setup_transport uses reference to key_pair internally so instead of cloning we can pass reference